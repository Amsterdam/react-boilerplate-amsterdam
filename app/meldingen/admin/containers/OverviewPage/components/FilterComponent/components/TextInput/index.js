import React from 'react';
import PropTypes from 'prop-types';

import { FieldControl } from 'react-reactive-form';

import './style.scss';

const TextInput = ({ name }) => (
  <FieldControl name={name} render={TextInputRender} />
);

TextInput.propTypes = {
  name: PropTypes.string
};

const TextInputRender = ({ handler, touched, hasError }) => (
  <div>
    <div className="rij mode_input text rij_verplicht">
      <div className="label">
        <label htmlFor="formInput">Id</label>
      </div>

      <div className="invoer">
        <input name="" id="formInput" value="" className="input" type="text" {...handler()} />
      </div>
      <div>
        {touched
          && hasError('required')
          && 'Id is required'}
      </div>

    </div>
  </div>
);

TextInputRender.propTypes = {
  handler: PropTypes.func.isRequired,
  touched: PropTypes.boolean,
  hasError: PropTypes.func.isRequired,
};

export default TextInput;
