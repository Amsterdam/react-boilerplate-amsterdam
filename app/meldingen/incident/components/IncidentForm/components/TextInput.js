import React from 'react';
import PropTypes from 'prop-types';

const TextInput = ({ handler, touched, hasError, meta }) => (
  <div>
    <input placeholder={meta.placeholder} {...handler()} />
    <span>
      {touched
      && hasError('required')
      && `${meta.label} is verplicht veld`}
      {touched
      && hasError('email')
      && `${meta.label} moet een geldig e-mailadres bevatten`}
    </span>
  </div>
);

TextInput.propTypes = {
  handler: PropTypes.func,
  touched: PropTypes.boolean,
  hasError: PropTypes.func,
  meta: PropTypes.object
};

export default TextInput;
