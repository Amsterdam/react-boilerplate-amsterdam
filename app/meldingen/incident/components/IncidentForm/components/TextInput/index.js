import React from 'react';
import PropTypes from 'prop-types';

import ErrorMessage from '../ErrorMessage/';

const TextInput = ({ handler, touched, hasError, meta }) => (
  <div>
    <input placeholder={meta.placeholder} {...handler()} />

    <ErrorMessage
      touched={touched}
      hasError={hasError}
    />
  </div>
);

TextInput.propTypes = {
  handler: PropTypes.func,
  touched: PropTypes.bool,
  hasError: PropTypes.func,
  meta: PropTypes.object
};

export default TextInput;
