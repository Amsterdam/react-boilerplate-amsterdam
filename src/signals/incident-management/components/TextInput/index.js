import React from 'react';
import PropTypes from 'prop-types';

import { FieldControl } from 'react-reactive-form';
// import withFieldControl from '../FieldControlWrapper';
import './style.scss';


export const TextInput = ({ name, control, ...props }) => (
  <div>
    <FieldControl name={name} control={control} render={TextInputRender(props)} />
  </div>);

TextInput.propTypes = {
  name: PropTypes.string.isRequired,
  control: PropTypes.object.isRequired
};


export const TextInputRender = (props) => {
  const { name, display, placeholder } = props;
  const render = ({ handler }) => (<div>
    <div className="mode_input text rij_verplicht">
      <div className="label">
        <label htmlFor={`form${name}`}>{display}</label>
      </div>

      <div className="invoer">
        <input name="" id={`form${name}`} value="" className="input" type="text" {...handler()} placeholder={placeholder} />
      </div>

    </div>
  </div>);

  render.defaultProps = {
    touched: false,
    placeholder: ''
  };

  render.propTypes = {
    handler: PropTypes.func.isRequired,
  };
  return render;
};

