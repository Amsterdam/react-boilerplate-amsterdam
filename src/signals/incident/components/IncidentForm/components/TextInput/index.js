import React from 'react';
import PropTypes from 'prop-types';

import Title from '../Title/';
import ErrorMessage from '../ErrorMessage/';
const TextInput = ({ handler, touched, hasError, meta, parent, getError, validatorsOrOpts }) => (
  <div>
    {meta && meta.isVisible ?
      <div className="row mode_input">
        <Title meta={meta} options={validatorsOrOpts} />

        <div className={`${meta.className || 'col-12'} invoer`}>
          <input
            type={meta.type}
            placeholder={meta.placeholder}
            onChange={(e) => meta.updateIncident && parent.meta.setIncident({ [meta.name]: e.target.value })}
            {...handler()}
          />
        </div>

        <ErrorMessage
          touched={touched}
          hasError={hasError}
          getError={getError}
        />
      </div>
       : ''}
  </div>
);

TextInput.propTypes = {
  handler: PropTypes.func,
  touched: PropTypes.bool,
  hasError: PropTypes.func,
  meta: PropTypes.object,
  parent: PropTypes.object,
  getError: PropTypes.func,
  validatorsOrOpts: PropTypes.object
};

export default TextInput;
