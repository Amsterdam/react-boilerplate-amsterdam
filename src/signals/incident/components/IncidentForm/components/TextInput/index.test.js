import React from 'react';
import { shallow } from 'enzyme';

import TextInput from './index';

describe('Form component <TextInput />', () => {
  let wrapper;
  let handler;
  let touched;
  let getError;
  let hasError;
  let parent;

  beforeEach(() => {
    handler = jest.fn();
    touched = false;
    getError = jest.fn();
    hasError = jest.fn();
    parent = {
      meta: {
        setIncident: jest.fn()
      }
    };

    wrapper = shallow(<TextInput
      handler={handler}
      parent={parent}
      touched={touched}
      hasError={hasError}
      getError={getError}
    />);
  });

  describe('rendering', () => {
    it('should render field correctly', () => {
      wrapper.setProps({
        meta: {
          name: 'input-field-name',
          type: 'text',
          placeholder: 'type here',
          ifVisible: true
        }
      });

      expect(handler).toHaveBeenCalledWith();
      expect(wrapper).toMatchSnapshot();
    });

    it('should render no text field when not visible', () => {
      wrapper.setProps({
        meta: {
          name: 'input-field-name',
          type: 'text',
          placeholder: 'type here',
          ifVisible: false
        }
      });

      expect(handler).not.toHaveBeenCalled();
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('events', () => {
    it('sets incident when value changes', () => {
      wrapper.setProps({
        meta: {
          name: 'input-field-name',
          type: 'text',
          placeholder: 'type here',
          ifVisible: true,
          updateIncident: true
        }
      });

      const event = { target: { value: 'diabolo' } };
      wrapper.find('input').simulate('change', event);

      expect(parent.meta.setIncident).toHaveBeenCalledWith({
        'input-field-name': 'diabolo'
      });
    });

    it('does nothing when updateIncident is false', () => {
      wrapper.setProps({
        meta: {
          name: 'input-field-name',
          type: 'text',
          placeholder: 'type here',
          ifVisible: true,
          updateIncident: false
        }
      });

      const event = { target: { value: 'diabolo' } };
      wrapper.find('input').simulate('change', event);

      expect(parent.meta.setIncident).not.toHaveBeenCalled();
    });
  });
});
