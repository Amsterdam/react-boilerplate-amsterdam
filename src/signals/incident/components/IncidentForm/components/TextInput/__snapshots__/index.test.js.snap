// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form component <TextInput /> rendering should render field correctly 1`] = `
<div>
  <div
    className="row mode_input"
  >
    <Title
      meta={
        Object {
          "ifVisible": true,
          "name": "input-field-name",
          "placeholder": "type here",
          "type": "text",
        }
      }
    />
    <div
      className="col-12 invoer"
    >
      <input
        onChange={[Function]}
        placeholder="type here"
        type="text"
      />
    </div>
    <ErrorMessage
      getError={[MockFunction]}
      hasError={[MockFunction]}
      touched={false}
    />
  </div>
</div>
`;

exports[`Form component <TextInput /> rendering should render no text field when not visible 1`] = `<div />`;
